compiler/src/iree/compiler/Codegen/Common/test/materialize_encoding_into_nop.mlir

```diff
@@ -5,8 +5,12 @@
 #map2 = affine_map<(d0, d1, d2) -> (d0, d1)>
 #encoding_lhs = #iree_encoding.encoding<operand_index = 0, op_type = matmul, element_types = [f32, f32, f32], user_indexing_maps = [#map, #map1, #map2]>
 func.func @pack_unpack_gemm_lhs(%arg0 : tensor<?x?xf32>) -> tensor<?x?xf32> {
+  %c0 = arith.constant 0 : index
+  %c1 = arith.constant 1 : index
+  %d0 = tensor.dim %arg0, %c0 : tensor<?x?xf32>
+  %d1 = tensor.dim %arg0, %c1 : tensor<?x?xf32>
   %0 = iree_encoding.set_encoding %arg0 : tensor<?x?xf32> -> tensor<?x?xf32, #encoding_lhs>
-  %1 = iree_encoding.unset_encoding %0 : tensor<?x?xf32, #encoding_lhs> -> tensor<?x?xf32>
+  %1 = iree_encoding.unset_encoding %0 : tensor<?x?xf32, #encoding_lhs> -> tensor<?x?xf32>{%d0, %d1}
```

Comment:
What are the chances we will end up doing the same for `set_encoding` ?

Reply:
Not at all.. I do not have a picture for that. The main issue is that there are no source to query the padded sizes. We could highly end up with the old solution (i.e., query_upper_bound), but that will mess up the HAL dep again. The HAL will depend on Codegen to get the storage size of a tensor. So it looks infeasible to me at this moment.